framework/README.md

```diff
@@ -0,0 +1,205 @@
+
+See [api_vulkan_sample.h](./api_vulkan_sample.h) and [api_vulkan_sample.cpp](./api_vulkan_sample.cpp).
+
+### Support for Vulkan-Hpp
+
+While the framework itself primarily uses the C-Interface for Vulkan, both the high level and the API sample base class also come with [Vulkan-Hpp](https://github.com/KhronosGroup/Vulkan-Hpp) variants, letting you write samples using the C++ Vulkan language bindings instead.
+
+See [hpp_vulkan_sample.h](./hpp_vulkan_sample.h) / [hpp_vulkan_sample.cpp](./hpp_vulkan_sample.cpp) and [hpp_api_vulkan_sample.h](./hpp_api_vulkan_sample.h) / [hpp_api_vulkan_sample.cpp](./hpp_api_vulkan_sample.cpp).
+
+## Commonly used framework concepts
+
+### Enabling extensions
+
+Vulkan is an extensible api. New features are usually exposed through either instance or device extensions. Extensions can be enabled in the constructor of both the high level and api base sample class:
+
+```cpp
+MySample::MySample()
+{
+    add_instance_extension(VK_SOME_INSTANCE_EXTENSION_NAME);
+	add_device_extension(VK_SOME_DEVICE_EXTENSION_NAME);
+}
+```
+
+**Note:** The framework uses the [Volk](https://github.com/zeux/volk) meta-loader, which will automatically load extension function pointers for all enabled extensions. There is no need to manually get extension pointer functions.
+
+### Changing the Vulkan api version
+
+By default all samples create a Vulkan 1.0 instance. Higher versions can be requested in the constructor of a sample:
+
+```cpp
+MySample::MySample()
+{
+    set_api_version(VK_API_VERSION_1_2);
+}
+```
+
+### Requesting GPU features
+
+Most extensions also require enabling accompanying. This can be done by overriding the `request_gpu_features` function of the base class:
+
+```cpp
+void MySample::request_gpu_features(vkb::PhysicalDevice &gpu)
+{
+    // Get a reference to the feature structure required for an extension
+	auto &requested_extension_feature = gpu
+    request_extension_features<VkPhysicalDeviceSomeExtensionFeaturesKHR>(VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SOME_EXTENSION_FEATURES_KHR);
+
+    // Enable a selected feature
+	requested_extension_feature.featureName = VK_TRUE;
+}
+```
+
+### Extending the graphical user interface
+
+The framework includes a graphical user interface based on [Dear ImGui](https://github.com/ocornut/imgui). This can be used by samples to display values and add controls like buttons, dropdowns, etc.
+
+To add additional elements to the UI of a sample, you override the respective function from the base class:
+
+Samples based on the high level base class need to override the `draw_gui` function:
+
+```cpp
+void MySample::draw_gui()
+{
+    if (ImGui::Checkbox("Enable Option", &option_enabled))
+    {
+        ...
+    }
+}
+```
+
+Samples based on the api base class need to override the `on_update_ui_overlay` function:
+
+```cpp
+void MyApiSample::on_update_ui_overlay(vkb::Drawer &drawer)
+{
+ 	if (drawer.checkbox("Enable option", &option_enabled))
+	{
+        ...
+	}
+}
+```
```

Comment:
What are the functional differences between `draw_gui` and `on_update_ui_overlay`?

Reply:
I'd love to add that, but that's actually not documented and since I didn't write the framework I probably couldn't explain. But then again this readme is meant to be a starting point, so I hope it's okay to be vague in certain spots.